### PR TITLE
apply tower best practices to services when cloning inner services

### PR DESCRIPTION
**Pull request by garypen**
When I was investigating connection resets I noted that we weren't following tower documentation with respect to [cloning inner services](https://docs.rs/tower/latest/tower/trait.Service.html#be-careful-when-cloning-inner-services). 

This PR addresses that. The change to the supergraph seems to be most invasive. I have run the tests manually many times and the change seems good, but probably worth closer examination.

### CODE REVIEW

**Review by bnjjj — APPROVED** [review]

File: NEXT_CHANGELOG.md [hunk 1]
```diff
@@ -28,11 +28,11 @@ By [@USERNAME](https://github.com/USERNAME) in https://github.com/apollographql/
 ## ❗ BREAKING ❗
 ## 🚀 Features
 
-### Add support for dhat based heap profiling [PR #XXXX](https://github.com/apollographql/router/pull/XXXX))
+### Add support for dhat based heap profiling [PR #1829](https://github.com/apollographql/router/pull/1829))
```
Comment (Geal) [inline NEXT_CHANGELOG.md:31]: this is the wrong changelog?

File: apollo-router/src/services/supergraph_service.rs [hunk 2]
```diff
@@ -87,19 +85,16 @@ where
     type Future = BoxFuture<'static, Result<Self::Response, Self::Error>>;
 
     fn poll_ready(&mut self, cx: &mut std::task::Context<'_>) -> Poll<Result<(), Self::Error>> {
-        // We need to obtain references to two hot services for use in call.
-        // The reason for us to clone here is that the async block needs to own the hot services,
-        // and cloning will produce a cold service. Therefore cloning in `SupergraphService#call` is not
-        // a valid course of action.
-        self.ready_query_planner_service
-            .get_or_insert_with(|| self.query_planner_service.clone())
+        self.query_planner_service
```
Comment (Geal) [inline apollo-router/src/services/supergraph_service.rs:88]: this is much cleaner :+1:

**Review by Geal — CHANGES_REQUESTED** [review]

File: NEXT_CHANGELOG.md (hunk 1 above)
Comment (garypen) [reply NEXT_CHANGELOG.md:31]: Yeah, I was just being lazy and fixing it here rather than raising a new PR...

Comment (garypen) [reply NEXT_CHANGELOG.md:31]: I wasn't going to put in a changelog entry for this change. It doesn't feel big enough to warrant mentioning. I can add one if you think it needs one.

Comment (Geal) [reply NEXT_CHANGELOG.md:31]: I think there should be one yes. It can be put in the maintenance section, it's not a functional change

Comment (garypen) [reply NEXT_CHANGELOG.md:31]: Done: https://github.com/apollographql/router/pull/2030/commits/2817f79e3851a6a7c24db8e56e62390474c61d78

**Review by Geal — APPROVED** [review]

**Pull request merged**